datastore/datastore/storagebackend/postgresql/postgresql.go

```diff
@@ -165,6 +166,16 @@ func NewPostgreSQL() (*PostgreSQL, error) {
 	setUpsertTSInsertCmd()
 	setUpsertTSSelectCmd()
 
+	// cleanup the database at regular intervals
+	ticker := time.NewTicker(cleanupInterval)
+	go func() {
+		for range ticker.C {
```

Comment:
I wonder what happens if the cleanup takes longer than the interval. Is the created channel buffered?

Not very practical in practice, I know...

Reply:
I checked it by triggering every 0.1 seconds, while the cleanup took 0.25 seconds. It just kept running a new cleanup straight away. Nothing blew up.

Reply:
Note that cleanups are now logged, so we can keep an eye on them. Maybe even add cleanups as a metric to be monitor them properly through prometheus/grafana.